src/main/java/org/fundacionjala/coding/kevinherrera/crypto/CesarCrypto.java

```diff
@@ -0,0 +1,78 @@
+package org.fundacionjala.coding.kevinherrera.crypto;
+
+/**
+ * Class to encode a message using a key.
+ */
+public abstract class CesarCrypto {
+
+    protected static final  int CHAR_Z_UPPER_HASHCODE = Character.hashCode('Z');
+    protected static final  int CHAR_A_UPPER_HASHCODE = Character.hashCode('A');
+
+    /**
+     * Enum encrypt tasks.
+     */
+    enum EncryptTask { ENCODE, DECODE }
+
+    /**
+     * encode a message using a Integer key.
+     * @param message message to encode
+     * @param key key to encode
+     * @return message encoded
+     */
+    protected String encode(final String message, final int key) {
+        return null;
+    }
+
+    /**
+     * decode a message using a Integer key.
+     * @param key key to decode
+     * @param encodeMessage message to decode
+     * @return message as lowercase type decoded
+     */
+    protected String decode(final int key, final String encodeMessage) {
+        return null;
+    }
+
+
+    /**
+     * encode a message using a String key.
+     * @param message message to encode
+     * @param key key string to encode
+     * @return message encoded
+     */
+    protected String encode(final String message, final String key) {
+        return null;
+    }
+
+    /**
+     * decode a message using a String key.
+     * @param key key string to decode
+     * @param encodeMessage message to decode
+     * @return message decoded
+     */
+    protected String decode(final String key, final String encodeMessage) {

```

Comment:
You should have only two abstracts methods.

abstract String encode(final String message, final String key);
abstract String decode(final String key, final String encodeMessage);

Reply:
for the integer key scenario
You can parse as int the String.

Reply:
done